### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by junaidshaikh-js**
This PR includes 

- Users can now login, signup, and logout from the app 
- Added loader 
- Added toast notifications for the login, signup, and logout state.

### CODE REVIEW

File: src/components/buttons/PrimaryOutlineBtn.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  children,
+  type = "button",
+  cnames,
+  onClick,
+  disable,
+}: buttonProps) => {
+  return (
+    <button
+      className={
+        "py-2 px-2.5 rounded text-black border border-primary hover:bg-primary hover:text-white transition-colors	 " +
```
Comment (hinanshisuthar) [inline src/components/buttons/PrimaryOutlineBtn.tsx:13]: What does '+' imply in classname? Got to know something new

**Review by hinanshisuthar — COMMENTED** [review]
- Clean code practices followed
- The toaster looks really cool
- Nice auth implementation
- Ready to merge

**Review by MonikaJhingan — APPROVED** [review]
The Ui looks good and the logic has been well.implemented as well as indentrd. Error handling is also done using try catch .It should be merged.

File: src/components/buttons/PrimaryOutlineBtn.tsx (hunk 1 above)
Comment (junaidshaikh-js) [reply src/components/buttons/PrimaryOutlineBtn.tsx:13]: I am just concatenating the strings there.

**Pull request merged**